src/Symfony/Component/HttpKernel/Attribute/MapQueryString.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+/**
+ * Controller parameter tag to map Query String to typed object and validate it.
+ *
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class MapQueryString
```

Comment:
It might be just me, but I feel the `String` suffix looks weird. Even though it's technically a Query String, using the `String` suffix feels like this resolves to a `string` while it does not.

Reply:
`MapQuery` or `MapQueryParameter` or ...?

Reply:
MapQueryString works for me, that's a common name for this